src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -113,12 +113,27 @@ class AuthScreens extends React.Component {
         // Load policies, maybe creating a new policy first.
         Linking.getInitialURL()
             .then((url) => {
-                if (this.shouldCreateFreePolicy(url)) {
-                    Policy.createAndGetPolicyList();
+                if (!url) {
                     return;
                 }
-
-                Policy.getPolicyList();
+                Navigation.isNavigationReady().then(() => {
```

Comment:
I'm curious what happens if we switched the order and placed `Linking.getInitialURL()` inside this callback. Would it work any better? I think it makes more logical sense to me that if I was just looking at this code with a fresh set of eyes, I would wonder "why are we getting the initial URL before navigation is ready?" 

I think it makes more sense that we wouldn't get the initial URL until after the navigation is ready.

Reply:
Also, I suggest moving the whole contents of this callback into a class method. This will help keep the code inside `componentDidMount()` to a minimum. A lot is happening there already.